### ISSUE TITLE
CustomerSignup with organization_type NULL

### ISSUE THREAD

**Issue by bart-at-mobicage** [our-city-app/oca-backend#752]
I'll manually add the organization type such that the user can continue...

`195.130.155.139 - - [22/Feb/2018:20:45:21 +0100] "POST /unauthenticated/osa/customer/signup HTTP/1.1" 200 129 https://dashboard.onzestadapp.be/customers/signup "Mozilla/5.0 (iPhone; CPU iPhone OS 11_2_5 like Mac OS X) AppleWebKit/604.5.6 (KHTML, like Gecko) Version/11.0 Mobile/15D60 Safari/604.1" "rogerthat-server.appspot.com" ms=702 cpu_ms=62 cpm_usd=9.1195e-8 loading_request=0 instance=00c61b117c844e4e84ae5dea8af743845ec36afa3827b784d00f5b6ed800a74b89f27a4a41120e64 app_engine_release=1.9.54 trace_id=-`

`[XX-OFFLOADv1]{"function":"/unauthenticated/osa/customer/signup","status":true,"request_data":{"city_customer_id":5733990728728576,"customer":{"city":"Lo-Reninge","name":"***","language":"nl","address1":"***","telephone":"***","user_email":"***","zip_code":"8647"},"company":{"website":"","city":"Lo-Reninge","name":"***","address1":"***","telephone":"***","vat":"***","organization_type":null,"facebook_page":"","user_email":"***","zip_code":"8647"},"recaptcha_token":"***"},"timestamp":1519328721.9898,"response_data":{"errormsg":null,"success":true},"type":"web"} (/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/lib/log_offload/log_offload.py:57)`

```
2018-02-22 21:11:36.273 CET
GET /customers/signup?data=***&email=***

2018-02-22 21:11:36.318 CET
Missing translation for organizationType None (/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/add_1_monkey_patches.py:127)
Traceback (most recent call last):
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1511, in __call__
    rv = self.handle_exception(request, response, e)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1505, in __call__
    rv = self.router.dispatch(request, response)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1253, in default_dispatcher
    return route.handler_adapter(request, response)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1077, in __call__
    return handler.dispatch()
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/shop/handlers.py", line 516, in dispatch
    return super(PublicPageHandler, self).dispatch()
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 547, in dispatch
    return self.handle_exception(e, self.app.debug)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 545, in dispatch
    return method(*args, **kwargs)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/shop/handlers.py", line 534, in get
    complete_customer_signup(email, data)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 164, in typechecked_return
    result = f(*args, **kwargs)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 142, in typechecked_f
    return f(**kwargs)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/shop/bizz.py", line 2763, in complete_customer_signup
    signup.inbox_message_key = _send_new_customer_signup_message(service_user, signup)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/shop/bizz.py", line 2637, in _send_new_customer_signup_message
    summary = get_signup_summary(sln_settings.main_language, customer_signup)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/shop/bizz.py", line 2604, in get_signup_summary
    org_type, lang, customer_signup.city_customer.app_id)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/rogerthat/models/__init__.py", line 895, in localized_singular_organization_type
    raise ValueError('Missing translation for organizationType %s' % organization_type)
ValueError: Missing translation for organizationType None
```

**Comment by bart-at-mobicage** [comment]
Maybe related

```
2018-02-22 10:05:33.702 CET
"POST /unauthenticated/osa/customer/signup HTTP/1.1" 500 225 https://dashboard.onzestadapp.be/customers/signup "Mozilla/5.0 (Windows NT 6.1; Trident/7.0; MAMD; rv:11.0) like Gecko" "rogerthat-server.appspot.com" ms=34 cpu_ms=59 cpm_usd=2.5144999999999998e-8 loading_request=0 instance=00c61b117ce135323bbf6e40755c1c46340204203fc755d100751f48a6d0b96b4ef879127ab8 app_engine_release=1.9.54 trace_id=-

2018-02-22 10:05:33.732 CET
Incorrect type received for parameter 'city_customer_id'. Expected (<type 'int'>, <type 'long'>) and got <type 'NoneType'> (None). (/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/add_1_monkey_patches.py:127)
Traceback (most recent call last):
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1511, in __call__
    rv = self.handle_exception(request, response, e)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1505, in __call__
    rv = self.router.dispatch(request, response)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1253, in default_dispatcher
    return route.handler_adapter(request, response)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 1077, in __call__
    return handler.dispatch()
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 547, in dispatch
    return self.handle_exception(e, self.app.debug)
  File "/base/alloc/tmpfs/dynamic_runtimes/python27/a7637d5531ec9deb_unzipped/python27_lib/versions/third_party/webapp2-2.3/webapp2.py", line 545, in dispatch
    return method(*args, **kwargs)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/restapi.py", line 229, in post
    params = parse_parameters(f, parameters)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 187, in parse_parameters
    return get_parameters(parameters, kwarg_types)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 211, in get_parameters
    for name, type_ in kwarg_types.iteritems()}
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 211, in <dictcomp>
    for name, type_ in kwarg_types.iteritems()}
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 285, in parse_parameter
    return _parse_value(name, raw_type, value)
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 403, in _parse_value
    raize()
  File "/base/data/home/apps/e~rogerthat-server/20180219t132013.407764961184727921/mcfw/rpc.py", line 398, in raize
    % (name, type_, type(value), value))
ValueError: Incorrect type received for parameter 'city_customer_id'. Expected (<type 'int'>, <type 'long'>) and got <type 'NoneType'> (None).

```

**Comment by bart-at-mobicage** [comment]
Another one

`[04/Mar/2018:14:46:05 +0100] "POST /unauthenticated/osa/customer/signup HTTP/1.1" 500 225 https://rogerth.at/customers/signup "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Mobile Safari/537.36" "rogerthat-server.appspot.com" ms=195 cpu_ms=448 cpm_usd=2.5144999999999998e-8 loading_request=0 instance=00c61b117ce3fa54621fcfb4766753d787ccd7c0e19d4f039e0c969814e5795ddcd8fb40b059 app_engine_release=1.9.54 trace_id=-`

**Comment by abom** [comment]
@bart-at-mobicage This last one was caused by me trying to reproduce the error, the only way i could reproduce it was by calling the method directly from the browser's javascript console!

**Comment by lucasvanhalst** [comment]
I'm guessing it happens when the /unauthenticated/osa/customer/org/types request fails or is very slow

**Comment by abom** [comment]
@lucasvanhalst Thanks!
I was wrong, in case of errors when getting organization types, the next button won't be disabled :(
It would be disabled on delays only.

**Comment by abom** [comment]
A possible fix for the organization type problem #813, although i couldn't find any errors when calling the rest api to get organization types.
